Update compose configuration

impca opened the pull request:
##### SUMMARY
Fix custom certificates not working in docker-compose configuration.

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
 - Installer

##### AWX VERSION
```
3.0.1
```

##### ADDITIONAL INFORMATION
When running awx via docker-compose and using custom certificates (for LDAP auth or whatever else...), update-ca-trust has to be called afer starting the container to actually use new certificates (just as it is called when using docker to run - https://github.com/ansible/awx/blob/devel/installer/roles/local_docker/tasks/standalone.yml#L119-L120 ). Withou this, LDAP auth with custom certs will fail even though certs are mounted correctly.

impca committed:
Update compose configuration

When running awx via docker-compose and using custom certificates (for LDAP auth or whatever else...), update-ca-trust has to be called afer starting the container to actually use new certificates (just as it is called when using docker to run - https://github.com/ansible/awx/blob/devel/installer/roles/local_docker/tasks/standalone.yml#L119-L120 ).

jakemcdermott commented on installer/roles/local_docker/tasks/compose.yml:
```diff
@@ -12,3 +12,6 @@
     project_src: "{{ docker_compose_dir }}"
+    
+- name: Update CA trust in awx_web container
+  command: docker exec awx_web_1 '/usr/bin/update-ca-trust'
```
In the standalone installer, we only run this command if the service starting task has a `changed` status to keep things idempotent. I think doing a similar check here would also be a good idea.

impca replied:
I see what you mean, I added status check.

jakemcdermott commented:
Hi. Thanks for contributing. I reviewed your PR and left some requested changes in the PR comments.

When you incorporate the changes, please also leave a note here on how you tested this and we'll have another look. Thank you!

impca committed:
update docker compose installer

Only run commands to update certs when config changes.

impca commented:
I added a check that will run the task when docker compose configuration changes or when the container starts. In the docker role, this is all part of a single step, but in the docker compose role, it's split into two steps:
1. First, config is written https://github.com/ansible/awx/blob/9add96a0d338f805f1e501083aa079cc8eea2e5e/installer/roles/local_docker/tasks/compose.yml#L7-L11 
2. Then containers are started: https://github.com/ansible/awx/blob/9add96a0d338f805f1e501083aa079cc8eea2e5e/installer/roles/local_docker/tasks/compose.yml#L13-L16

Furthermore, it is apparently not enough to update certs in the web container. According to https://github.com/ansible/awx/issues/3010 certs should be updated in both awx_web and awx_task containers. So I added that, too. https://github.com/ansible/awx/blob/9add96a0d338f805f1e501083aa079cc8eea2e5e/installer/roles/local_docker/tasks/compose.yml#L18-L24 

I tested it by manually editing the docker-compose.yaml and by stopping the docker services. That forced the installer to overwrite the changed docker-compose.yaml config, start the services and run the corresponding tasks.

jakemcdermott commented:
Nicely done, @impca. Thanks.
